### DISCUSSION TITLE
Microsoft.Extensions.AI - What is it?

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/semantic-kernel#9163] [Q&A] [answered]
I had a look at the diagram on the [Microsoft.Extensions.AI post](https://devblogs.microsoft.com/dotnet/introducing-microsoft-extensions-ai-preview/) and it mentions SK.

![image](https://github.com/user-attachments/assets/449d8ff0-d75e-445e-9e9f-83334c4acb70)

But none of these packages in https://github.com/dotnet/extensions seem to use SK.

![image](https://github.com/user-attachments/assets/b1444d37-b35b-4c4b-83bc-69e3dd9ecf3a)

So, is Microsoft.Extensions.AI something you use _instead_ of SK? If not, are these any samples that show how this can be used with SK?

The samples in https://github.com/dotnet/ai-samples/blob/main/src/microsoft-extensions-ai/README.md don't seem to refer SK either.

![image](https://github.com/user-attachments/assets/0c768483-5f62-4fc2-893a-4b15d8749884)

**Comment by evchaki** [comment, marked as answer]
@rajyraman - thanks for asking.  We have a blog post here ( https://devblogs.microsoft.com/semantic-kernel/microsoft-extensions-ai-simplifying-ai-integration-for-net-partners/ ) that might be helpful. 

Microsoft.Extensions.AI is made from Microsoft.SemanticKernel.Abstractions (https://www.nuget.org/packages/Microsoft.SemanticKernel.Abstractions) in order to help support our growing ISV and AI Service partner community.  When Microsoft.Extensions.AI is GA we will use that as an SDK that we call into - just like we do for all of the other SDK's below SK.

Let me know if you have any other questions.

**Reply by rajyraman** [reply]
@evchaki - I just finished watching Steve Sanderson's demo on dotnetConf about this. Amazing stuff! Makes total sense now.

**Comment by AshD** [comment]
Is there a Github link for Microsoft.Extensions.AI?

**Reply by AshD** [reply]
Found it https://github.com/dotnet/extensions/tree/main/src/Libraries/Microsoft.Extensions.AI
